### PR TITLE
XdsClient: remove now-unnecessary XdsCertificateProviderPluginMapInterface

### PR DESCRIPTION
**Pull request by markdroth**
I added this interface in #30506, but it is no longer needed with the approach taken in #30759.

### CODE REVIEW

File: src/core/ext/xds/xds_bootstrap_grpc.cc [hunk 1]
```diff
@@ -461,8 +464,17 @@ std::string GrpcXdsBootstrap::ToString() const {
     parts.push_back("      },\n");
   }
   parts.push_back("}");
```
Comment (yashykt) [inline src/core/ext/xds/xds_bootstrap_grpc.cc:466]: This seems to want a '\n' at the end

File: src/core/ext/xds/xds_bootstrap_grpc.cc [hunk 2]
```diff
@@ -376,24 +367,36 @@ grpc_error_handle GrpcXdsBootstrap::ParseCertificateProvider(
...
-        }
+        config_json = it->second;
       }
     } else {
-      // "config" is an optional field, so create an empty JSON object.
-      absl::Status status = certificate_provider_plugin_map_->AddPlugin(
-          instance_name, plugin_name, Json::Object());
-      if (!status.ok()) {
-        error_list.push_back(absl_status_to_grpc_error(status));
+      // "config" is an optional field, so use an empty JSON object.
```
Comment (yashykt) [inline src/core/ext/xds/xds_bootstrap_grpc.cc:381]: maybe a better comment would be - "config" is an optional field that defaults to an empty JSON object.

**Review by yashykt — APPROVED** [review]
Approving but minor comments

File: src/core/ext/xds/xds_bootstrap_grpc.cc (hunk 1 above)
Comment (markdroth) [reply src/core/ext/xds/xds_bootstrap_grpc.cc]: Good catch, fixed.

**Pull request merged**